process/sync/storageBootstrap/shardStorageBootstrapper.go

```diff
@@ -0,0 +1,154 @@
+package storageBootstrap
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/process/sync"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+type shardStorageBootstrapper struct {
+	*storageBootstrapper
+	miniBlocksResolver dataRetriever.MiniBlocksResolver
+}
+
+// NewShardStorageBootstrapper is method used to create a nes storage bootstrapper
+func NewShardStorageBootstrapper(arguments ArgsStorageBootstrapper) (*shardStorageBootstrapper, error) {
+	base := &storageBootstrapper{
+		bootStorer:       arguments.BootStorer,
+		forkDetector:     arguments.ForkDetector,
+		blkExecutor:      arguments.BlockProcessor,
+		blkc:             arguments.ChainHandler,
+		marshalizer:      arguments.Marshalizer,
+		store:            arguments.Store,
+		shardCoordinator: arguments.ShardCoordinator,
+
+		uint64Converter:     arguments.Uint64Converter,
+		bootstrapRoundIndex: arguments.BootstrapRoundIndex,
+	}
+
+	miniBlocksResolver, err := arguments.ResolversFinder.IntraShardResolver(factory.MiniBlocksTopic)
+	if err != nil {
+		return nil, err
+	}
+
+	miniBlocksRes, ok := miniBlocksResolver.(dataRetriever.MiniBlocksResolver)
+	if !ok {
+		return nil, process.ErrWrongTypeAssertion
+	}
+
+	boot := shardStorageBootstrapper{
+		storageBootstrapper: base,
+		miniBlocksResolver:  miniBlocksRes,
+	}
+
+	base.bootstrapper = &boot
+	hdrNonceHashDataUnit := dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(boot.shardCoordinator.SelfId())
+	base.headerNonceHashStore = boot.store.GetStorer(hdrNonceHashDataUnit)
+
+	return &boot, nil
+}
+
+// LoadFromStorage will load all blocks from storage
+func (ssb *shardStorageBootstrapper) LoadFromStorage() error {
+	return ssb.loadBlocks()
+}
+
+// IsInterfaceNil returns true if there is no value under the interface
+func (ssb *shardStorageBootstrapper) IsInterfaceNil() bool {
+	return ssb == nil
+}
+
+func (ssb *shardStorageBootstrapper) getHeader(hash []byte) (data.HeaderHandler, error) {
+	return ssb.getShardHeaderFromStorage(hash)
```

Comment:
why do we need another wrapper over the same functionality provided by the same implementation.

Reply:
removed function getShardHeaderFromStorage